core/templates/dev/head/pages/exploration_player/AudioControlsDirective.js

```diff
@@ -64,12 +65,19 @@ oppia.directive('audioControls', [
           };
 
           $scope.playPauseAudioTranslation = function() {
-            // TODO(tjiang11): Change from on-demand loading to pre-loading.
-
-            // TODO(tjiang11): On first play, ask learner to pick language
-            // and subsequently for confirmation to use bandwidth 
-            // to download audio files.
+            if (!AudioPreloaderService.hasPreloaded()) {
+              if (getCurrentAudioTranslation()) {
```

Comment:
Ah, interesting. On reading I had interpreted "getCurrentAudioTranslation()" as "get the audio translation that's currently playing". I wonder if we could change the name to make the intended meaning clearer...

And yep that sounds good! In the non-preload case, what about subsequent downloads, though? Will there be a modal each time?